scripts/publish-circleci.js

```diff
@@ -0,0 +1,121 @@
+ *
+ * Instructions:
+ * Run this script with SALESFORCEDX_VSCODE_VERSION, CIRCLECI_TOKEN & CIRCLECI_BUILD as environment variables
+ * i.e. SALESFORCEDX_VSCODE_VERSION=x.y.z ./scripts/publish.js
+ *
+ */
+
+// Checks that you are running this with Node v12.4.0 and above
+const [version, major, minor, patch] = process.version.match(
+  /^v(\d+)\.?(\d+)\.?(\*|\d+)$/
+);
+if (parseInt(major) !== 12 || parseInt(minor) < 4) {
+  console.log(
+    'You do not have the right version of node. We require version 12.4.0.'
+  );
+  process.exit(-1);
+}
+// Check if you have installed all the required tooling
+if (!shell.which('vsce')) {
+  console.log('vsce is not installed or could not be found');
+  process.exit(-1);
+}
+
+if (!shell.which('lerna')) {
+  console.log('lerna is not installed or could not be found');
+  process.exit(-1);
+}
+
+if (!shell.which('aws')) {
+  console.log('aws cli is not installed or could not be found');
+  process.exit(-1);
+}
+
+// Checks that you have access to our bucket on AWS
+const awsExitCode = shell.exec(
+  'aws s3 ls s3://dfc-data-production/media/vscode/SHA256.md',
+  {
+    silent: true
+  }
+).code;
+if (awsExitCode !== 0) {
+  console.log(
+    'You do not have the s3 command line installed or you do not have access to the aws s3 bucket.'
+  );
+  process.exit(-1);
+}
+
+// Checks that you have access to the salesforce publisher
+const publishers = shell
+  .exec('vsce ls-publishers', { silent: true })
+  .stdout.trim();
+if (!publishers.includes('salesforce')) {
+  console.log(
+    'You do not have the vsce command line installed or you do not have access to the salesforce publisher id as part of vsce.'
+  );
+  process.exit(-1);
+}
+
+// Checks that you have specified the next version as an environment variable, and that it's properly formatted.
+const nextVersion = process.env['SALESFORCEDX_VSCODE_VERSION'];
+const releaseBranchName = `release/v${nextVersion}`;
+if (!nextVersion) {
+  console.log(
+    'You must specify the next version of the extension by setting SALESFORCEDX_VSCODE_VERSION as an environment variable.'
+  );
+  process.exit(-1);
+} else {
+  const [version, major, minor, patch] = nextVersion.match(
+    /^(\d+)\.(\d+)\.(\d+)$/
+  );
+  const currentBranch = shell
+    .exec('git rev-parse --abbrev-ref HEAD', {
+      silent: true
+    })
+    .stdout.trim();
+
+  if (!currentBranch !== releaseBranchName) {
+    console.log(
+      `You must execute this script in a release branch, you are currently running the script on branch ${currentBranch}`
+    );
+    process.exit(-1);
+  }
+}
+
+// Download vsix files from CircleCI
+shell.exec('./scripts/download-vsix-from-circleci.js');
+
+// Generate the SHA256 and append to the file
+shell.exec(`npm run vscode:sha256`);
+
+// Concatenate the contents to the proper SHA256.md
+shell.exec('./scripts/concatenate-sha256.js');
+
+// Remove the temp SHA256 file
+shell.rm('./SHA256');
+
```

Comment:
What do you think about a **--dryrun** option that performs all of the local operations, but doesn't commit and publish?

Reply:
Is that because we want to be able to test the publish script or something different ?

Reply:
Yes, for testing.

Reply:
Let's talk more about that and I can address it as a follow up PR. I think having a dryrun option in this script means to exclusively download the vsix files in the correct location, which I don't think would be a good way to test it and will make the script more complex.

Reply:
I added a new npm script called `circleci:artifacts` that just downloads the vsix files in the correct location. Would that be enough to address this ?

Reply:
Yes it does, thanks.